lib/decoder.ex

```diff
@@ -64,262 +65,284 @@ defmodule Jason.Decoder do
   defp string_decode_function(%{strings: :copy}), do: &:binary.copy/1
   defp string_decode_function(%{strings: :reference}), do: &(&1)
 
-  defp value(data, original, skip, stack, key_decode, string_decode) do
+  defp float_decode_function(%{floats: :decimals}) do
+    fn string, token, skip ->
+      try do
+        Decimal.new(string)
```

Comment:
`decimal` is an optional dependency in mix.exs. Do we need to make it non-optional? Otherwise, I think this should be wrapped in `if Code.ensure_loaded?(Decimal) do ... end` conditionals: https://hexdocs.pm/elixir/Code.html#ensure_loaded?/1

Reply:
Code.ensure_loaded? is expensive, so I don’t think it should be added or it should be done at compilation time. :)